lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -303,9 +303,19 @@ def get(self, request, *args, **kwargs):  # pylint: disable=too-many-statements
             )
             available_seq_ids = {str(usage_key) for usage_key in user_course_outline.sequences}
 
+            available_section_ids = {str(section.usage_key) for section in user_course_outline.sections}
+
+            # course_blocks is a reference to the root of the course,
+            # so we go through the chapters (sections) and keep only those
+            # which are part of the outline.
+            course_blocks['children'] = [chapter_data
+                                         for chapter_data in course_blocks['children']
+                                         if chapter_data['id'] in available_section_ids
+                                         ] if 'children' in course_blocks else []
```

Comment:
Nit: Please keep the old `.get('children', [])` piece instead of adding the `if 'children' in...` bit at the tail end of the list comprehension. Try to limit the use of the `foo = bar if something else bar2` syntax to simple one-liners.

So something like:

```python
            course_blocks['children'] = [
                chapter_data
                for chapter_data in course_blocks.get()'children', [])
                if chapter_data['id'] in available_section_ids
            ]
```

FWIW, the linter is probably might complain to you about the indentation here. I'm not sure if we have it in the rules for list comprehensions, but I tend to follow our [long function call conventions](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines) when doing them.

Reply:
Sure! Fixed.

Reply:
I also fixed the CI pipeline errors.